src/scripts/createBlock.ts

```diff
@@ -92,30 +103,39 @@ async function getBlockOperations(): Promise<Operation[]> {
   );
 
   let i = 0;
-  const blockchainOperations = unappliedFirestoreOps.map(op =>
-    buildBlockchainOpFromFirestoreOp(op, i++)
+  const blockchainOperations = await Promise.all(
+    unappliedFirestoreOps.map(op => buildBlockchainOpFromFirestoreOp(op, i++))
   );
   return blockchainOperations.filter(x => x) as Operation[];
 }
 
 /**
  * Assumes the blockchain list is sorted by sequence number. Returns the last sequence number + 1.
  */
-function getNextBlockSequenceNumber(blockchain: Block[]): number {
-  return blockchain[blockchain.length - 1].data.sequence + 1;
+function getNextBlockSequenceNumber(blockchain: VirtualBlock[]): number {
+  if (blockchain.length > 0) {
+    return blockchain[blockchain.length - 1].data.sequence + 1;
+  } else {
+    return 0;
+  }
 }
 
 /**
  * Assumes the blockchain list is sorted by sequence number. Returns the hash of the last block.
  */
-function getLastBlockHash(blockchain: Block[]): string {
-  return blockchain[blockchain.length - 1].metadata.multiHash;
+function getLastBlockHash(blockchain: VirtualBlock[]): string {
+  if (blockchain.length > 0) {
+    return blockchain[blockchain.length - 1].metadata.multiHash;
+  } else {
+    return "no_previous_blocks_found";
```

Comment:
I'd prefer if this were a data type that is different than that of a multihash, like a unique symbol, but not that important